backend/app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -1,47 +1,39 @@
 class Api::V1::BookmarksController < ApplicationController
-    before_action :set_user
+    before_action :check_login
 
     def create
-        begin
-            knowledge = Knowledge.find(bookmark_params[:knowledge_id].to_i)
-        rescue ActiveRecord::RecordNotFound
-            return render json: { errors: '対象のデータが見つかりません' }, status: 404
-        end
+        knowledge = Knowledge.find(bookmark_params[:knowledge_id].to_i)
         
-        if knowledge.bookmarks.create(user_id: @user.id)
-            render json: { knowledge: knowledge.as_json().merge(isBookmark: true) }, status: 200
-        else
-            render json: { errors: knowledge.errors }, status: 422
-        end
+        bookmark = knowledge.bookmarks.new(user: current_api_v1_user)
+        
+        render json: { errors: knowledge.errors.full_message }, status: 422 and return if bookmark.invalid?
+
+        bookmark.save!
+
+        render json: { knowledge: knowledge.as_json.merge(isBookmark: true) }, status: 200
+    rescue ActiveRecord::RecordNotFound
+        return render json: { errors: '対象のデータが見つかりません' }, status: 404
+    rescue StandardError => e
+        render json: { errors: e.message }, status: 500
     end
 
     def destroy
-        begin
-            knowledge = Knowledge.find(params[:id].to_i)
-        rescue ActiveRecord::RecordNotFound
-            return render json: { error: '対象のデータが見つかりません' }, status: 404
-        end
-
-        if knowledge.bookmarks.where(knowledge_id: knowledge.id).destroy_all
-            bookmark = knowledge.bookmarks.where(knowledge_id: knowledge.id).first
-            
-            render json: { knowledge: knowledge.as_json().merge(isBookmark: bookmark.present?) }, status: 200
-        else
-            render json: { errors: knowledge.errors }, status: 422
-        end
-    end
+        knowledge = Knowledge.find(params[:id].to_i)
 
-    private
+        knowledge.bookmarks.where(user_id: params[:user_id]).destroy_all
+
+        bookmark = knowledge.bookmarks.where(knowledge_id: knowledge.id).first
```

Comment:
以下でknowledgeに紐ずくbookmarksを取得しており、もうすでに絞り込めんているので、再度whereで絞り込む必要はありません！

```suggestion
        bookmark = knowledge.bookmarks.first
```

Reply:
knowledge_idで絞り込むのではなく、user_idで絞り込んで削除されているか確認する処理に変更しました